Test-case: Access properties from Dictionary variable

niklasHagner opened the pull request:
There is no support for Dictionaries except for when used as `esi:foreach collection` 

My test scenario...
```
        <esi:assign name="pizzaIngredients" value="{'cheese': 'true', 'avocado': 'false'}" />
        <esi:assign name="hasAvocado" value="$(pizzaIngredients{avocado})" />
        <esi:vars>
          <p>Does a pizza have avocado: $(hasAvocado)</p>
        </esi:vars>
```
..results in unparsed code:
```
<p>hasCheese: '$(pizzaIngredients{cheese})</p>
``` 

## The problem
The issue is that lexer.js can't handle property accessors such as: `$(pizzaIngredients{avocado})` 

`lexer.js` will abort with "Unknown keyword"!  
```
if (keywords.indexOf(token.cargo) === -1) {
        token.abort("Unknown keyword", token.cargo);
 }
```
At this point `token.cargo` would be `avocado` so it would throw `Unknown keyword 'avocado'`

_The only known keywords are `[matches, matches_i, has, has_i]`_

## Possible solution ⚒️  

I've got a hacky solution which works for my particular test case
https://github.com/BonnierNews/local-esi/compare/master...niklasHagner:dictionary-hack?expand=1

Basicall: removes `token.abort("Unknown keyword", token.cargo)` and assumes unknown tokens are attempts to access a property.

jonaswalden reviewed (commented):
Seems to work with quotes around the key when accessing the value `$(pizzaIngredients{'cheese'})` instead of `$(pizzaIngredients{cheese})`.

Unquoted keys are however used throughout the examples in the ESI specification so it is a bug that it doesn't work here.

niklasHagner commented:
Aight. I've updated the test case to use single-quotes around the key.

So in the end this PR will not add any logic, but the test-case in itself is a good addition.

I added a comment to clarify that Akamai ESI supports unquoted keys, but local-esi doesn't. Not sure if you're totally anti-comments here 😺  - if so I could clarify it some other way.
(Perhaps a test-case where I expect `$(pizzaIngredients{cheese})` to not be parsed incorrectly)

niklasHagner commented:
👍  https://github.com/BonnierNews/local-esi/commit/313b6b2928d07b5cdd47ac66bbb2ced950342f9b
